cmd/notation/plugin.go

```diff
@@ -62,3 +109,160 @@ func listPlugins(command *cobra.Command) error {
 	}
 	return tw.Flush()
 }
+
+func installPlugin(command *cobra.Command, args []string, force bool) error {
+	if len(args) != 1 {
+		return errors.New("missing plugin package")
+	}
+
+	pluginSrcPath := args[0]
+
+	pluginBinary := filepath.Base(pluginSrcPath)
+
+	// get plugin metadata
+	cmd := exec.Command(pluginSrcPath, "get-plugin-metadata")
+
+	output, err := cmd.Output()
+	if err != nil {
+		return err
+	}
+
+	var newPlugin map[string]interface{}
+	err = json.Unmarshal([]byte(output), &newPlugin)
+	if err != nil {
+		return err
+	}
+
+	pluginName := newPlugin["name"].(string)
+	newPluginVersion := newPlugin["version"].(string)
+	newSemVersion, err := semver.NewVersion(newPluginVersion)
+	if err != nil {
+		return err
+	}
+
+	// get plugin directory
+	pluginDir, err := dir.PluginFS().SysPath(pluginName)
+	if err != nil {
+		return err
+	}
+
+	// Check if plugin directory exists
+	_, err = os.Stat(pluginDir)
+
+	// create the directory, if not exist
+	if os.IsNotExist(err) {
+		if err := os.MkdirAll(pluginDir, 0755); err != nil {
+			return err
+		}
+	}
+
+	pluginDestPath := pluginDir + "/" + pluginBinary
+
+	// Check if plugin exists
+	_, err = os.Stat(pluginDestPath)
+
+	// copy plugin, if not exist
+	if os.IsNotExist(err) {
+		copyPlugin(pluginSrcPath, pluginDestPath)
+	}
+
+	// overwrite plugin, if force flag is set
+	if err == nil && force {
+		fmt.Printf("Overwriting plugin %s in directory %s\n", pluginBinary, pluginDir)
+		copyPlugin(pluginSrcPath, pluginDestPath)
+	}
+
+	// if plugin exists and force flag is not set, get plugin metadata
+	if err == nil && !force {
+		cmd := exec.Command(pluginDestPath, "get-plugin-metadata")
+
+		output, err := cmd.Output()
+		if err != nil {
+			return err
+		}
+
+		var currentPlugin map[string]interface{}
+		err = json.Unmarshal([]byte(output), &currentPlugin)
+		if err != nil {
+			return err
+		}
+
+		// check if new plugin version is greater than current plugin version
+		currentPluginVersion := currentPlugin["version"].(string)
+		currentVersion, err := semver.NewVersion(currentPluginVersion)
+		if err != nil {
+			return err
+		}
+
+		// copy plugin, if new plugin version is greater than current plugin version
+		if newSemVersion.GreaterThan(currentVersion) {
```

Comment:
can we use `golang.org/x/mod/semver` instead of "github.com/Masterminds/semver"? 
Because `golang.org/x/mod/semver` was used by notation-go, we will not add an additional dependency.

Reply:
I certainly can, I've made the necessary changes to use `golang.org/x/mod/semver'. :)